apollo-router/src/warp_http_server_factory.rs

```diff
@@ -676,4 +692,36 @@ mod tests {
             .iter()
             .for_each(|accepts| assert!(!prefers_html(accepts.to_string())));
     }
+
+    #[test(tokio::test)]
+    async fn it_provides_health_status_body() -> Result<(), FederatedServerError> {
+        let filter = get_health_request();
+
+        let res = warp::test::request()
+            .path("/.well-known/apollo/server-health")
+            .reply(&filter)
+            .await;
+
+        assert_eq!(res.status(), 200);
+        assert_json_snapshot!(String::from_utf8_lossy(res.body()));
```

Comment:
[nit] Same here. You can probably assert the entire response using assert_debug_snapshot